front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -15,9 +15,71 @@ import type {
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
 import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+import type { ArrayElement } from 'utils/types';
 
 import { electricalProfilesDesignValues } from './consts';
 
+type SpeedLimitTagValue = ArrayElement<SimulationResponseSuccess['mrsp']['values']>;
+const getTag = (source?: SpeedLimitTagValue['source']): { name: string; color: string } => {
+  let name: string;
+  let color: string;
+
+  if (!source) {
+    name = 'missing_from_train';
+    color = '#94918E';
+  } else {
+    switch (source.speed_limit_source_type) {
+      case 'given_train_tag':
+        name = source.tag || 'missing_from_train';
+        color = '#494641';
+        break;
+      case 'fallback_tag':
+        name = source.tag || 'missing_from_train';
+        color = '#216482';
+        break;
+      case 'unknown_tag':
+        name = 'incompatible';
+        color = '#EAA72B';
+        break;
+      default:
+        name = 'missing_from_train';
+        color = '#94918E';
```

Comment:
We use two times these lines (when no source and default case), maybe we could simplify like this (if there is no source it goes to default) : 

```suggestion
const getTag = (source?: SpeedLimitTagValue['source']): { name: string; color: string } => {
  let name: string;
  let color: string;

  switch (source?.speed_limit_source_type) {
    case 'given_train_tag':
      name = source.tag || 'missing_from_train';
      color = '#494641';
      break;
    case 'fallback_tag':
      name = source.tag || 'missing_from_train';
      color = '#216482';
      break;
    case 'unknown_tag':
      name = 'incompatible';
      color = '#EAA72B';
      break;
    default:
      name = 'missing_from_train';
      color = '#94918E';
  }

  return { name, color };
};
```

If you don't like the fact that we still check all of `source_types` if there is no `source`, we can do like this : 

```suggestion
const getTag = (source?: SpeedLimitTagValue['source']): { name: string; color: string } => {
  let name = 'missing_from_train';
  let color = '#94918E';

  if (source) {
    name = source.speed_limit_source_type === 'unknown_tag' ? 'incompatible' : source.tag;

    if (source.speed_limit_source_type === 'given_train_tag') {
      color = '#494641';
    } else if (source.speed_limit_source_type === 'fallback_tag') {
      color = '#216482';
    } else if (source.speed_limit_source_type === 'unknown_tag') {
      color = '#EAA72B';
    }
  }

  return { name, color };
};
```